File: submissions/0007-reverse-integer/solution.java
class Solution {
public int reverse(int x) {
    int ans=0, temp=x, rem;
    while(temp!=0)
    {
        rem=temp%10;
        temp/=10;
        if(ans > Integer.MAX_VALUE/10 || (ans == Integer.MAX_VALUE/10 && rem>7))
            return 0;
        if(ans < Integer.MIN_VALUE/10 || (ans == Integer.MIN_VALUE/10 && rem<-8))
            return 0;
        ans=ans*10+rem;
        
    }
    return ans;
    }
};

File: submissions/0007-reverse-integer/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        sign = -1 if x<0 else 1
        # rev = int(str(abs(x))[::-1]) * sign
        return (int(str(abs(x))[::-1]) * sign) if (-2**31)<(int(str(abs(x))[::-1]) * sign)<((2**31)-1) else 0
